### ISSUE TITLE
avatars are not shown

### ISSUE THREAD

**Issue by silberzwiebel** [ForNeVeR/kaiwa#136]
I only see my own avatar but now avatars from my contacts. Different from #90, avatars are also not working in the single chat. I'm using a self-hosted prosody nightly server with mod_vcard and mod_pep_vcard_avatar enabled.

**Comment by vitalyster** [comment]
You can set `localStorage.debug` flag to `true` in browser console before login, and then you will see detailed XMPP XML dump. Is there any error about vcards?

**Comment by silberzwiebel** [comment]
Thanks for the quick reply. I've done that and see a lot of output but no errors about vcards...

**Comment by vitalyster** [comment]
did you see vcard request and responses?

**Comment by silberzwiebel** [comment]
No. I searched for 'vcard' and the only thing that is in the log then is
this one:

<iq type='result'
to='someone@somewhere/98cf7333-893d-4090-a0a8-cf16c5c76534'
from='somewhere' id='5b3975b6-c376-4f4f-9585-9e7bc6f80758'
xmlns='jabber:client'><query
xmlns='http://jabber.org/protocol/disco#info'><identity type='pep'
name='Prosody' category='pubsub'/><identity type='im' name='Prosody'
category='server'/><feature var='urn:xmpp:blocking'/><feature
var='http://jabber.org/protocol/commands'/><feature
var='http://jabber.org/protocol/pubsub#publish'/><feature
var='msgoffline'/><feature var='jabber:iq:version'/><feature
var='urn:xmpp:http:upload'/><feature var='jabber:iq:register'/><feature
var='vcard-temp'/><feature var='jabber:iq:roster'/><feature
var='urn:xmpp:time'/><feature var='jabber:iq:time'/><feature
var='jabber:iq:private'/><feature var='jabber:iq:last'/><feature
var='http://prosody.im/protocol/vcard-pep-integration'/><feature
var='urn:xmpp:carbons:2'/><feature var='urn:xmpp:carbons:1'/><feature
var='urn:xmpp:mam:0'/><feature var='urn:xmpp:ping'/><feature
var='http://jabber.org/protocol/disco#info'/><feature
var='http://jabber.org/protocol/disco#items'/><x type='result'
xmlns='jabber:x:data'><field type='hidden'
var='FORM_TYPE'><value>urn:xmpp:http:upload</value></field><field
type='text-single'
var='max-file-size'><value>10485760</value></field></x></query></iq>

**Comment by vitalyster** [comment]
do you have mod_pep enabled ?

**Comment by silberzwiebel** [comment]
You mean on the server side, right?
Yes, I do. OMEMO encryption works for Conversations and afaik this also
depends on mod_pep.

**Comment by vitalyster** [comment]
so, you should receive avatar notifications like [this](http://xmpp.org/extensions/xep-0084.html#process-subnotify) - do you see that in XMPP log?

**Comment by silberzwiebel** [comment]
Yes, I have stuff like this in the log. But it looks to me as if it only
fetches my own avatar (which is correctly shown in Kaiwa), because the
id= is always the same and I only have one time the avatar:data with a
large data block.

**Comment by vitalyster** [comment]
are your contacts on the same server, or from different servers? Maybe different server doesn't notify you about avatars

**Comment by silberzwiebel** [comment]
Most of them are on the same server.
